frontend/packages/knative-plugin/src/components/sink-source/__tests__/SinkSourceModal.spec.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import {
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory/modal';
+import { ResourceDropdownField } from '@console/shared';
+import SinkSourceModal from '../SinkSourceModal';
+
+type SinkSourceModalProps = React.ComponentProps<typeof SinkSourceModal>;
+let formProps: SinkSourceModalProps;
+
+describe('SinkSourceModal Form', () => {
+  const formValues = {
+    sink: {
+      ref: {
+        apiVersion: 'serving.knative.dev/v1',
+        kind: 'Service',
+        name: 'event-greeter',
+      },
+    },
+  };
+  beforeEach(() => {
+    formProps = {
+      values: formValues,
+      namespace: 'myapp',
+      resourceName: 'myappes',
+      errors: {},
+      touched: {},
+      isSubmitting: true,
+      isValidating: true,
+      status: {},
+      submitCount: 0,
+      dirty: false,
+      handleReset: jest.fn(),
+      handleSubmit: jest.fn(),
+      getFieldProps: jest.fn(),
+      handleBlur: jest.fn(),
+      handleChange: jest.fn(),
+      initialErrors: {},
+      initialStatus: {},
+      initialTouched: {},
+      isValid: true,
+      registerField: jest.fn(),
+      resetForm: jest.fn(),
+      setErrors: jest.fn(),
+      setFieldError: jest.fn(),
+      setFieldTouched: jest.fn(),
+      setFieldValue: jest.fn(),
+      setFormikState: jest.fn(),
+      setStatus: jest.fn(),
+      setSubmitting: jest.fn(),
+      setTouched: jest.fn(),
+      setValues: jest.fn(),
+      submitForm: jest.fn(),
+      unregisterField: jest.fn(),
+      validateField: jest.fn(),
+      validateForm: jest.fn(),
+      getFieldMeta: jest.fn(),
+      validateOnBlur: true,
+      validateOnChange: true,
+      initialValues: formValues,
+    };
+  });
+  it('should render ModalTitle, body and footer', () => {
+    const eventSourceForm = shallow(<SinkSourceModal {...formProps} />);
+    expect(eventSourceForm.find(ModalTitle)).toHaveLength(1);
+    expect(eventSourceForm.find(ModalBody)).toHaveLength(1);
+    expect(eventSourceForm.find(ModalSubmitFooter)).toHaveLength(1);
+  });
+  it('should render ResourceDropdownField for service', () => {
+    const eventSourceForm = shallow(<SinkSourceModal {...formProps} />);
```

Comment:
Can we have some more tests for submit action buttons and move this `shallow(<SinkSourceModal {...formProps} />` to beforeEach function.